src/main/java/com/example/MyBatisDemo/controller/StudentController.java

```diff
@@ -44,7 +45,7 @@ public StudentDisplayDto addStudent(@Valid @RequestBody StudentAdditionDto stude
     }
 
     @PostMapping("/{id}/subjects")
-    public StudentSubjectsDisplayDto assignSubjectsToStudent(@Valid@PathVariable(value = "id")Long id, @RequestBody StudentSubjectsAdditionDto subjectsAdditionDto){
+    public StudentSubjectsDisplayDto assignSubjectsToStudent(@Valid@PathVariable(value = "id")Long id, @RequestBody StudentSubjectsAdditionDto subjectsAdditionDto) throws ResourceNotFoundException {
```

Comment:
It is generally preferred to throw unchecked exceptions. If you guys are not aware of checked and unchecked exceptions please do read about it. It is a very important and basic concept.

Reply:
yes